ui/lib/extensions/src/react/modal-extension.tsx

```diff
@@ -10,7 +10,7 @@ export const ModalExtension = () => {
   React.useLayoutEffect(() => {
     const controller = new AbortController();
     window.addEventListener(
-      'popstate',
+      'single-spa:routing-event',
```

Comment:
I'm not sure if this should depend on a custom event. What was the issue with popstate?

Reply:
Initially it was set up with `single-spa:routing-event` I changed it to `popstate` thinking the same that this shouldn't depend on custom event. :D I am just preserving the initial intent by having `single-spa:routing-event`. Otherwise I prefer `popstate` as well.

Reply:
I just changed it to `popstate`